### ISSUE TITLE
Add support for incoming XML streams and non-reconnecting client factories.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2617]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)| @ralphm reported|
|-|-|
|Trac ID|trac#2617|
|Type|enhancement|
|Created|2007-05-01 12:34:26Z|

Currently, Xish and the Jabber support in Words assumes outgoing streams. For example, the `XmlStreamFactory` in both derives from `ReconnectingClientFactory`.

<details><summary>Searchable metadata</summary>

```
trac-id__2617 2617
type__enhancement enhancement
reporter__ralphm ralphm
priority__highest highest
milestone__Words_0_6 Words-0.6
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1178022866000000 1178022866000000
changetime__1178538402000000 1178538402000000
version__None None
owner__ 
cc__ralphm cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_1"></a>|@ralphm commented|
|-|-|

One way to solve this is to put the setting up of bootstrap event observers in a mixin and then have the current `XmlStream` classes derive from that mixin and `ReconnectingClientFactory`. Then implementations can reuse the mixin together with plain `ClientFactory` or `ServerFactory`.

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_2"></a>|@ralphm removed owner|
|-|-|

Implemented mixin in [branches/xish-xmlstream-mixin-2617.](branches/xish-xmlstream-mixin-2617.)

Please review.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @ralphm|
|-|-|

I think that the XmlStreamFactory in jabber can still inherit from the XmlStreamFactory in xish, just overriding its init method. Also, a bunch of docstrings are missing (the Dummy classes in tests, buildProtocol...). Finally, can you break the for loop in buildProtocol in two lines?

Thanks.

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_4"></a>|@ralphm set owner to @therve|
|-|-|

Processed review comments. Please review again?

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve set owner to @ralphm|
|-|-|

Great, please remove the protocol import from jabber.xmlstream, then merge.

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_6"></a>|@ralphm set status to `closed`|
|-|-|

(In [20203]) Create XmlStreamFactoryMixin and derive factories from it for better reuse.

Author: ralphm
Reviewer: therve
Fixes [#2617](https://github.com/twisted/twisted/issues/2617).

This change allows for the creation of server side factories and
non-reconnecting clients.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
